### DISCUSSION TITLE
AWS Lambda endpoint only show gray scale image

### DISCUSSION THREAD

**Discussion by giswqs** [developmentseed/titiler#549] [Q&A] [answered]
I deployed TiTler to Amazon Lambda using https://serverlessrepo.aws.amazon.com/applications/us-east-1/552819999234/TiTiler and got the endpoint URL. However, the image rendered using my own endpoint only shows gray scale image, while the demo TiTiler endpoint (https://titiler.xyz/) can show a color image. Any advice? 

My endpiont
![image](https://user-images.githubusercontent.com/5016453/204897858-886b3fbc-cd2c-402c-b0be-8e356f8ed91e.png)

TiTiler demo endpoint
![image](https://user-images.githubusercontent.com/5016453/204897714-dd9f546b-cfa6-41af-9456-41e492bc4441.png)

**Comment by giswqs** [comment]
I think I found the problem. Using the demo endpoint, `cog/statistics` works, but my own endpoint returns `{'detail': 'Not Found'}`. Is it because the AWS template uses an older version of TiTiler that does not support `cog/statistics`?

```python
endpoint= 'https://titiler.xyz' 
r = requests.get(
f"{endpoint}/cog/statistics",
params={
    "url": url,
},
).json()
r
```

**Comment by giswqs** [comment]
It turns out that the Amazon Application Registry uses TiTiler v0.3.1 while the demo endpoint uses v0.7.0. 

v0.3.1 does not support `/cog/statistics`. 

The next step is to figure out how to update TiTiler on AWS. I am new to Amazon Lambda. Still learning. 

![image](https://user-images.githubusercontent.com/5016453/204905377-31eb1f70-22ec-4777-8599-ba7fcb95e02a.png)

![image](https://user-images.githubusercontent.com/5016453/204905440-fcfee121-c857-4a5e-8919-8205ea311e98.png)

**Reply by vincentsarago** [reply]
see https://github.com/lambgeo/titiler-layer/issues/2

**Reply by vincentsarago** [reply]
@giswqs I'll try to publish a new version tonight

**Reply by giswqs** [reply]
@vincentsarago Nice! I look forward to the new version. Maybe also update Python 3.8 to 3.9 or 3.10.

**Reply by vincentsarago** [reply]
last time I checked python 3.10 was still not available in AWS Lambda. I'll do 3.9 for now

**Reply by giswqs** [reply]
Sounds good. Thanks.

**Reply by giswqs** [reply]
You are right. As of now, Python 3.10 is still not supported yet https://github.com/aws/aws-lambda-base-images/issues/31

**Reply by vincentsarago** [reply]
lambda-layer and SAM application are now up to date!

**Reply by vincentsarago** [reply]
https://github.com/lambgeo/titiler-layer#sam-application

**Reply by giswqs** [reply]
Awesome! Thank you. I will try it out

**Comment by giswqs** [comment, marked as answer]
@vincentsarago The [new version](https://github.com/lambgeo/titiler-layer#sam-application) works like a charm. Thank you.
